### PR TITLE
Django Templates, HTMX, and Tailwind Refactor

### PR DESCRIPTION
**Pull request by nathanroberts55**
The refactor was to replace the ReactJS frontend to now use the built-in Django Templates, Tailwindcss, and HTMX.

This also includes a change that replaces the OpenLibrary API (was experiencing a DDOS attack at the time of development) with the Google Book API (which has been more stable/consistent, but has a rate limit currently).

Currently has the API and Frontend apps still in the repo, want to confirm that the new frontend functions properly before removing the code for those.

### CODE REVIEW

File: BigABookClub/urls.py [hunk 1]
```diff
@@ -29,12 +31,14 @@
...
+    path("", views.index, name="index"),
+    path("books/", include("books.urls")),
+    path("goals/", include("goals.urls")),
+    path("accounts/", include("accounts.urls")),
+    path("__reload__/", include("django_browser_reload.urls")),
     path("", include("social_django.urls", namespace="social")),
-    path("", include("frontend.urls")),
-    path("api/", include("api.urls")),
+    # path("", include("frontend.urls")),
+    # path("api/", include("api.urls")),
```
Comment (dylhack) [inline BigABookClub/urls.py:41]: remove after testing

File: accounts/templates/accounts/profile.html [hunk 2]
```diff
@@ -1,64 +1,25 @@
...
+    <!-- #region Reading Resolution -->
+    {% include 'partials/_reading_goal.html' %}
+    <!-- #endregion Reading Resolution -->
+    <section class="flex flex-col mx-auto my-10">
+        <p class="mb-8 text-4xl font-bold text-center">Your Bookmarks</p>
+        {% if books %}
+        {% include 'partials/_recommendation_list.html' %}
+        {% endif %}
+    </section>
+    <script src="{% static '/js/scroll_position.js' %}"></script>
```
Comment (dylhack) [inline accounts/templates/accounts/profile.html:23]: should this be in the head?

File: accounts/templates/partials/_read_book_list.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    {% if goal.books_read.all %}
+    {% for book in goal.books_read.all %}
+    {% include 'partials/_read_book.html' with book=book %}
+    {% if not forloop.last %}
+    <hr />
+    {% endif %}
+    {% endfor %}
+    {% else %}
+    <p>No Books Read, yet. Keep Going!</p>
+    {% endif %}
```
Comment (dylhack) [inline accounts/templates/partials/_read_book_list.html:11]: no indentations?

File: accounts/templates/partials/_reading_goal.html [hunk 4]
```diff
@@ -0,0 +1,142 @@
+{% load static %}
+{% if goal %}
+<!-- Reading Resolution -->
+<section id="reading_goal_section" class="flex flex-col max-md:w-10/12 mx-auto my-10 space-y-12">
+    <div class="flex flex-col bg-[#6f42c166] rounded-3xl">
```
Comment (dylhack) [inline accounts/templates/partials/_reading_goal.html:5]: I see this color used in other places, should we consider adding it to tailwind's config as a shared color?

File: accounts/templates/partials/_reading_goal.html [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            <p class="text-5xl font-bold my-2 mb-5">Book Club Reading Resolution!</p>
+            <p>Get Smarter Everyday by joining the the Book Club Reading Resolution! The Book Club Reading Resolution is
+                a self-paced reading program designed to make reading fun, interactive, supportive. Set a goal for the
+                year and track each book you read, whether its with the book club or on your own! Each book you read
+                will earn yourself a 'Little A' sticker to go on a printable PDF Certificate and track your progress in
+                your profile in the Big A Book Club website. Reach your reading goal by the end of the year and earn a
+                special prize!</p>
+            <button hx-get="{% url 'create_goal' %}" hx-target="#reading_goal_section" hx-swap="outerHTML"
+                class="btn btn-wide btn-large btn-outline btn-neutral my-6 mx-auto text-lg">Start Reading
+                Resolution</button>
```
Comment (dylhack) [inline accounts/templates/partials/_reading_goal.html:137]: +1 if you could [incorporate localization](https://docs.djangoproject.com/en/5.1/topics/i18n/) for best practices

File: accounts/views.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    logout(request=request)
+    # messages.success(request, "You have successfully logged out.")
+    return redirect("/")
+
+
+# Create your views here.
+@login_required
+def account_profile(request) -> None:
+
```
Comment (dylhack) [inline accounts/views.py:24]: nit: empty line, could we lint / format?

File: accounts/views.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+from django.contrib.auth.decorators import login_required
+from books.models import Book
+from goals.models import ReadingGoal
+
+
+def account_logout(request) -> None:
+    context = {}
+
+    logout(request=request)
+    # messages.success(request, "You have successfully logged out.")
```
Comment (dylhack) [inline accounts/views.py:17]: nit: useless comment

File: books/forms.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
+from django.utils.translation import gettext_lazy as _
+from captcha.fields import ReCaptchaField
+from captcha.widgets import ReCaptchaV2Checkbox
+from django import forms
+from .models import Book
+
+
+class BookForm(forms.ModelForm):
+    # captcha = ReCaptchaField(widget=ReCaptchaV2Checkbox)
```
Comment (dylhack) [inline books/forms.py:9]: nit: useless comment

File: books/forms.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+    def clean_atrioc_streamlink(self):
+        cleaned_data = self.cleaned_data
+
+        if (
+            "source" in cleaned_data
+            and cleaned_data["source"] == Book.Sources.ATRIOC
+            and "stream_link" in cleaned_data
+            and cleaned_data["stream_link"] == ""
+        ):
```
Comment (dylhack) [inline books/forms.py:71]: these conditions are a little nasty, could we shorten this, and in other places too?

File: books/templates/partials/_currently_reading.html [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+            <div class="flex flex-row max-md:flex-col-reverse gap-4 items-center">
+                {% if cr_description %}
+                <p>{{ cr_description }}</p>
+                {% endif %}
+                {% if cr_image_url %}
+                <img class="w-[250px] mb-8" src="{{ cr_image_url }}"
+                    alt="{{currently_reading.title}} by {{currently_reading.author}} Currently Reading Book Cover Image">
+                {% endif %}
+            </div>
+            <a href="https://discord.com/channels/1159391121999413310/1173808374266212412" target="_blank"
```
Comment (dylhack) [inline books/templates/partials/_currently_reading.html:18]: this channel link is used in a couple places, could we put this in a constant?

**Review by dylhack — COMMENTED** [review]
a few nitpicks

File: accounts/templates/accounts/profile.html (hunk 2 above)
Comment (nathanroberts55) [reply accounts/templates/accounts/profile.html:23]: I only need it on specific pages, specifically the account and recommendation pages. So I only load it on those pages

File: accounts/templates/partials/_read_book_list.html (hunk 3 above)
Comment (nathanroberts55) [reply accounts/templates/partials/_read_book_list.html:11]: brother, I wish the linter/formatter did that, but supposedly the proper formatting is with not indentations.

File: accounts/templates/partials/_reading_goal.html (hunk 4 above)
Comment (nathanroberts55) [reply accounts/templates/partials/_reading_goal.html:5]: Good call

File: accounts/templates/partials/_reading_goal.html (hunk 5 above)
Comment (nathanroberts55) [reply accounts/templates/partials/_reading_goal.html:137]: Will address in new PR, please create an issue to track for future development

File: accounts/templates/partials/_reading_goal.html (hunk 4 above)
Comment (nathanroberts55) [reply accounts/templates/partials/_reading_goal.html:5]: Addressed in next commit

File: accounts/views.py (hunk 7 above)
Comment (nathanroberts55) [reply accounts/views.py:17]: Addressed in next commit

File: books/forms.py (hunk 8 above)
Comment (nathanroberts55) [reply books/forms.py:9]: Addressed in next commit

File: books/templates/partials/_currently_reading.html (hunk 10 above)
Comment (nathanroberts55) [reply books/templates/partials/_currently_reading.html:18]: Will address in future PR.

File: books/forms.py (hunk 9 above)
Comment (nathanroberts55) [reply books/forms.py:71]: Will Address in Future PR

File: BigABookClub/urls.py (hunk 1 above)
Comment (nathanroberts55) [reply BigABookClub/urls.py:41]: Yes sir, Mr. Hack Sir! (0_0)7

**Pull request merged**